tests/benchmark/conftest.py

```diff
@@ -7,6 +7,17 @@
 
 def pytest_collection_modifyitems(config, items):
     """Add the `benchmark` marker to all tests under `./tests/benchmark`."""
+    marker_expr = config.getoption("-m", default="")
+    run_benchmarks = marker_expr and (
+        "benchmark" in marker_expr and "not benchmark" not in marker_expr
+    )
+
+    items_to_remove = []
     for item in items:
```

Comment:
Let's use the trick from #1936 and use indexes when removing the items, it's significantly faster.